app/Models/Traits/UsesLinkedFields.php

```diff
@@ -339,13 +347,14 @@ public function cleanConditionalAnswers(Form $form): void
 
         foreach ($childQuestions as $child) {
             if (in_array($child->parent_id, $conditionalsToBeCleaned)) {
-                $fieldConfig = data_get($fieldsConfig, $question->linked_field_key);
+                $fieldConfig = data_get($fieldsConfig, $child->linked_field_key);
                 $property = data_get($fieldConfig, 'property', null);
                 if ($this->isPlainField($child->input_type)) {
-                    $entityProps[$property] = $this->getDefaultValue($child->input_type);
+                    $entityProps[$property] = null;
```

Comment:
The default value for a hidden child question should always be null, not a faked "empty" value. This is especially important for number fields where seeing a null vs a 0 in the exported data can have very different connotations.